src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.PerAnalyzerState.cs

```diff
@@ -346,14 +428,8 @@ public void OnCompilationEventGenerated(CompilationEvent compilationEvent, Analy
                     {
                         if (actionCounts.SyntaxTreeActionsCount > 0)
                         {
-                            var trees = compilationEvent.Compilation.SyntaxTrees;
-                            var map = new Dictionary<SyntaxTree, AnalyzerStateData>(trees.Count());
```

Comment:
@basoundr This should address the large number of dictionary entries we saw in perf traces.